src/frontend/components/SchedulingFields/index.tsx

```diff
@@ -0,0 +1,205 @@
+  startingAtDateTextLabel: {
+    defaultMessage: 'Starting date',
+    description:
+      'Label of the input field to add/update the start date of the scheduled event',
+    id: 'component.SchedulingFields.startingAtDateTextLabel',
+  },
+  startingAtTimeTextLabel: {
+    defaultMessage: 'Starting time',
+    description:
+      'Label of the input field to add/update the start time of the scheduled event',
+    id: 'component.SchedulingFields.startingAtTextLabel',
+  },
+  estimatedDurationTextLabel: {
+    defaultMessage: 'Estimated duration',
+    description:
+      'Label of the input field to add/update the estimated duration of the scheduled event',
+    id: 'component.SchedulingFields.estimatedDurationTextLabel',
+  },
+});
+
+// build a list of times from 09h00 to 19h30
+const _time = Duration.fromObject({ hours: 9 });
+const allTimesSuggestions = Array.from({ length: 22 }, (_v, k) => k * 30).map(
+  (minutes) => _time.plus({ minutes }).toFormat('hh:mm'),
+);
+
+const displayedDuration = (duration: Duration) => {
+  return duration.toFormat('h:mm');
+};
+
+const returnedDuration = (durationString: string) => {
+  if (durationString.length === 4) durationString = '0' + durationString;
+  return Duration.fromISOTime(durationString).toFormat('hh:mm:ss');
+};
+
+// build a list of durations from 15 minutes to 5 hours
+const _duration = Duration.fromObject({ minutes: 15 });
+const allDurationSuggestions = Array.from(
+  { length: 20 },
+  (_v, k) => k * 15,
+).map((minutes) => displayedDuration(_duration.plus({ minutes })));
+
+interface SchedulingFieldsProps {
+  startingAt: Nullable<string>;
+  estimatedDuration: Nullable<string>;
+  onStartingAtChange?: (date: string) => void;
+  onEstimatedDurationChange?: (duration: string) => void;
+  margin?: MarginType;
+}
+
+export const SchedulingFields = ({
+  startingAt,
+  estimatedDuration,
+  onStartingAtChange = () => {},
+  onEstimatedDurationChange,
+  margin,
+}: SchedulingFieldsProps) => {
+  const intl = useIntl();
+  Settings.defaultLocale = intl.locale;
+  Settings.defaultZone = 'utc';
+
+  const [startingAtDate, setStartingAtDate] = React.useState(
+    startingAt
+      ? DateTime.fromISO(startingAt)
+          .set({ hour: 0, minute: 0, second: 0 })
+          .toISO()
+      : '',
+  );
+  const [timeSuggestions, setTimeSuggestions] =
+    React.useState(allTimesSuggestions);
+  const [_startingAtTime, setStartingAtTime] = React.useState(
+    startingAt
+      ? DateTime.fromISO(startingAt).toLocaleString(DateTime.TIME_24_SIMPLE)
+      : '',
+  );
+  const [durationSuggestions, setDurationSuggestions] = React.useState(
+    allDurationSuggestions,
+  );
+  const [_estimatedDuration, setEstimatedDuration] = React.useState(
+    estimatedDuration
+      ? displayedDuration(Duration.fromISOTime(estimatedDuration))
+      : '',
+  );
+
+  const onStartingAtDateChange = (event: { value: string | string[] }) => {
+    setStartingAtDate(event.value as string);
```

Comment:
I’ m not a fan of this cast 😬

Reply:
Me neither…
I'll look for something more elegant.

Reply:
I would rather  check if the value is an array (with  `isArray` and handle it) or be able to use the value as a string without casting it

Reply:
WDYT ? https://github.com/openfun/marsha/pull/1445/commits/01f8e18c5c5eea70dc6b55dfbc384216d561c60d

Reply:
maybe check array is not empty ?